pkg/api/types.go

```diff
@@ -1895,6 +1895,9 @@ type DeleteOptions struct {
 	// The value zero indicates delete immediately. If this value is nil, the default grace period for the
 	// specified type will be used.
 	GracePeriodSeconds *int64 `json:"gracePeriodSeconds"`
+
+	// Only the object has the same UID will be deleted.
+	UID *types.UID `json:"uid,omitempty"`
```

Comment:
I think I want something that's like `Preconditions *struct { UID *types.UID ... }`. We need to share this struct with UpdateOptions (which I guess don't exist yet).